src/util/triplea/MapXMLCreator/TerritoryConnectionsPanel.java

```diff
@@ -0,0 +1,256 @@
+			g.setColor(Color.red);
+			g.drawString(centerName, Math.max(0, x_text_start), item.y + 5);
+		}
+		g.setColor(Color.red);
+	}
+	
+	protected void paintPreparation(final Map<String, Point> centers)
+	{
+		if (centers != null && !MapXMLHelper.s_territoyConnections.isEmpty())
+			return;
+		final Map<String, List<Area>> territoryAreas = new HashMap<String, List<Area>>();
+		for (final String territoryName : s_polygons.keySet())
+		{
+			final List<Polygon> listOfPolygons = s_polygons.get(territoryName);
+			final List<Area> listOfAreas = new ArrayList<Area>();
+			for (final Polygon p : listOfPolygons)
+			{
+				listOfAreas.add(new Area(p));
+			}
+			territoryAreas.put(territoryName, listOfAreas);
+
+		}
+		final String lineWidth = JOptionPane.showInputDialog(null, "Enter the width of territory border lines on your map? \r\n(eg: 1, or 2, etc.)", "1");
+		if (lineWidth == null)
+			return;
+		int scalePixels = 8;
+		double minOverlap = 32;
+		try
+		{
+			final int lineThickness = Integer.parseInt(lineWidth);
+			scalePixels = lineThickness * 4;
+			minOverlap = scalePixels * 4;
+		} catch (final NumberFormatException ex)
+		{
+		}
+		if (JOptionPane.showConfirmDialog(null, "Scale set to " + scalePixels + " pixels larger, and minimum overlap set to " + minOverlap + " pixels. \r\n"
+					+ "Do you wish to continue with this? \r\nSelect Yes to continue, Select No to override and change the size.", "Scale and Overlap Size", JOptionPane.YES_NO_OPTION) == 1)
+		{
+			final String scale = JOptionPane.showInputDialog(null,
+						"Enter the number of pixels larger each territory should become? \r\n(Normally 4x bigger than the border line width. eg: 4, or 8, etc)", "4");
+			if (scale == null)
+				return;
+			try
+			{
+				scalePixels = Integer.parseInt(scale);
+			} catch (final NumberFormatException ex)
+			{
+			}
+			final String overlap = JOptionPane.showInputDialog(null,
+						"Enter the minimum number of overlapping pixels for a connection? \r\n(Normally 16x bigger than the border line width. eg: 16, or 32, etc.)", "16");
+			if (overlap == null)
+				return;
+			try
+			{
+				minOverlap = Integer.parseInt(overlap);
+			} catch (final NumberFormatException ex)
+			{
+			}
+		}
+		
+		MapXMLHelper.s_territoyConnections.clear();
+		System.out.print("Now Scanning for Connections ... ");
+		// sort so that they are in alphabetic order (makes xml's prettier and easier to update in future)
+		final List<String> allTerritories = s_polygons == null ? new ArrayList<String>() : new ArrayList<String>(s_polygons.keySet());
+		Collections.sort(allTerritories, new AlphanumComparator());
+		final List<String> allAreas = new ArrayList<String>(territoryAreas.keySet());
+		Collections.sort(allAreas, new AlphanumComparator());
+		for (final String territory : allTerritories)
+		{
+			final LinkedHashSet<String> thisTerritoryConnections = new LinkedHashSet<String>();

```

Comment:
Has to be linkedHashSet due to static variable definition where it is added later on.